src/functions-and-arrays.js

```diff
@@ -1,73 +1,480 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(numOne, numTwo) {
+  if (numOne > numTwo) {
+    return numOne;
+  } else if (numTwo > numOne) {
+    return numTwo;
+  } else {
+    return numOne;
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(arr) {
+  if (arr.length == 0) {
+    return null;
+  } else if (arr.length == 1) {
+    return arr[0];
+  }
+
+  var longestWord = "";
+  for (var i = 0; i < arr.length; i++) {
+    if (longestWord.length < arr[i].length) {
+      longestWord = arr[i];
+    }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(nums) {
+  if (nums.length == 0) {
+    return 0;
+  } else if (nums.length == 1) {
+    return nums[0];
+  }
+  var sum = 0;
+  for (var i = 0; i < nums.length; i++) {
+    sum += nums[i];
+  }
+  if (sum === 0) {
+    return 0;
+  } else {
+    return sum;
+  }
+}
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(avgNums) {
+  if (avgNums.length == 0) {
+    return null;
+  } else if (avgNums.length == 1) {
+    return avgNums[0];
+  }
+  var avg = 0;
+  for (var i = 0; i < avgNums.length; i++) {
+    avg += avgNums[i];
+  }
+  avg = avg / avgNums.length;
+  avg = Math.sqrt(avg * avg);
+  return avg;
+}
+
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
+];
 
+function averageWordLength(avgWords) {
+  if (avgWords.length == 0) {
+    return null;
+  } else if (avgWords.length == 1) {
+    return avgWords[0].length;
+  }
+  var sum = 0;
```

Comment:
Try to be careful when naming your variables, "sum" doesn't make sense in this function does it?